### PR TITLE
fix(components): [table-v2] calculate the width of MainTable in the same way

### LINKED ISSUE #13520: [Component] [table-v2] 虚拟表格固定列时宽度计算有误
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.31`
- Element Plus Version: `2.3.0`
- Browser / OS: `'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 Edg/114.0.1823.67`
- Build Tool: `Vite`

### Related Component

- `el-table-v2`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdGFibGUtdjJcbiAgICA6Y29sdW1ucz1cImNvbHVtbnNcIlxuICAgIDpkYXRhPVwiZGF0YVwiXG4gICAgOnNvcnQtYnk9XCJzb3J0QnlcIlxuICAgIDp3aWR0aD1cIjcwMFwiXG4gICAgOmhlaWdodD1cIjQwMFwiXG4gICAgZml4ZWRcbiAgICBAY29sdW1uLXNvcnQ9XCJvblNvcnRcIlxuICAvPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IFRhYmxlVjJGaXhlZERpciwgVGFibGVWMlNvcnRPcmRlciB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuaW1wb3J0IHR5cGUgeyBTb3J0QnkgfSBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmNvbnN0IGdlbmVyYXRlQ29sdW1ucyA9IChsZW5ndGggPSAxMCwgcHJlZml4ID0gJ2NvbHVtbi0nLCBwcm9wcz86IGFueSkgPT5cbiAgQXJyYXkuZnJvbSh7IGxlbmd0aCB9KS5tYXAoKF8sIGNvbHVtbkluZGV4KSA9PiAoe1xuICAgIC4uLnByb3BzLFxuICAgIGtleTogYCR7cHJlZml4fSR7Y29sdW1uSW5kZXh9YCxcbiAgICBkYXRhS2V5OiBgJHtwcmVmaXh9JHtjb2x1bW5JbmRleH1gLFxuICAgIHRpdGxlOiBgQ29sdW1uICR7Y29sdW1uSW5kZXh9YCxcbiAgICB3aWR0aDogMTUwLFxuICB9KSlcblxuY29uc3QgZ2VuZXJhdGVEYXRhID0gKFxuICBjb2x1bW5zOiBSZXR1cm5UeXBlPHR5cGVvZiBnZW5lcmF0ZUNvbHVtbnM+LFxuICBsZW5ndGggPSAyMDAsXG4gIHByZWZpeCA9ICdyb3ctJ1xuKSA9PlxuICBBcnJheS5mcm9tKHsgbGVuZ3RoIH0pLm1hcCgoXywgcm93SW5kZXgpID0+IHtcbiAgICByZXR1cm4gY29sdW1ucy5yZWR1Y2UoXG4gICAgICAocm93RGF0YSwgY29sdW1uLCBjb2x1bW5JbmRleCkgPT4ge1xuICAgICAgICByb3dEYXRhW2NvbHVtbi5kYXRhS2V5XSA9IGBSb3cgJHtyb3dJbmRleH0gLSBDb2wgJHtjb2x1bW5JbmRleH1gXG4gICAgICAgIHJldHVybiByb3dEYXRhXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICBpZDogYCR7cHJlZml4fSR7cm93SW5kZXh9YCxcbiAgICAgICAgcGFyZW50SWQ6IG51bGwsXG4gICAgICB9XG4gICAgKVxuICB9KVxuXG5jb25zdCBjb2x1bW5zID0gZ2VuZXJhdGVDb2x1bW5zKDEwKVxubGV0IGRhdGEgPSBnZW5lcmF0ZURhdGEoY29sdW1ucywgMjAwKVxuXG5jb2x1bW5zWzBdLmZpeGVkID0gdHJ1ZVxuY29sdW1uc1sxXS5maXhlZCA9IFRhYmxlVjJGaXhlZERpci5MRUZUXG5jb2x1bW5zWzldLmZpeGVkID0gVGFibGVWMkZpeGVkRGlyLlJJR0hUXG5cbmZvciAobGV0IGkgPSAwOyBpIDwgMzsgaSsrKSBjb2x1bW5zW2ldLnNvcnRhYmxlID0gdHJ1ZVxuXG5jb25zdCBzb3J0QnkgPSByZWY8U29ydEJ5Pih7XG4gIGtleTogJ2NvbHVtbi0wJyxcbiAgb3JkZXI6IFRhYmxlVjJTb3J0T3JkZXIuQVNDLFxufSlcblxuY29uc3Qgb25Tb3J0ID0gKF9zb3J0Qnk6IFNvcnRCeSkgPT4ge1xuICBkYXRhID0gZGF0YS5yZXZlcnNlKClcbiAgc29ydEJ5LnZhbHVlID0gX3NvcnRCeVxufVxuPC9zY3JpcHQ+XG4ifQ==)

### Steps to reproduce

把滚动条拉到最右边，除了被固定的列的最后一列会有部分被隐藏，是固定列挡住了，大约是6px的大小。

### What is Expected?

固定列不会挡住除了被固定的列的最后一列

### What is actually happening?

把滚动条拉到最右边，除了被固定的列的最后一列会有部分被隐藏，是固定列挡住了，大约是6px的大小。

### Additional comments

_(empty)_

image

### PR DESCRIPTION
**Pull request by makedopamine**
closes #13520

### CODE REVIEW

File: packages/components/table-v2/src/table-v2.tsx [hunk 1]
```diff
@@ -144,6 +144,7 @@ const TableV2 = defineComponent({
         fixedData,
         estimatedRowHeight,
         bodyWidth: unref(bodyWidth),
+        bodyWidth: unref(bodyWidth) + vScrollbarSize,
```
Comment (btea) [inline packages/components/table-v2/src/table-v2.tsx:147]: two same properties? 🤔

Comment (makedopamine) [reply packages/components/table-v2/src/table-v2.tsx:147]: Sorry, it was careless of me. 😅

**Review by btea — APPROVED** [review]

**Review by ryuhangyeong — COMMENTED** [review]
@btea 

Can you provide a minimal example of the problem under what circumstances? 🙇

**Comment by btea** [comment]
@ryuhangyeong 
[before](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXRhYmxlLXYyXG4gICAgOmNvbHVtbnM9XCJjb2x1bW5zXCJcbiAgICA6ZGF0YT1cImRhdGFcIlxuICAgIDpzb3J0LWJ5PVwic29ydEJ5XCJcbiAgICA6d2lkdGg9XCI3MDBcIlxuICAgIDpoZWlnaHQ9XCI0MDBcIlxuICAgIGZpeGVkXG4gICAgQGNvbHVtbi1zb3J0PVwib25Tb3J0XCJcbiAgLz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgeyBUYWJsZVYyRml4ZWREaXIsIFRhYmxlVjJTb3J0T3JkZXIgfSBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmltcG9ydCB0eXBlIHsgU29ydEJ5IH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5jb25zdCBnZW5lcmF0ZUNvbHVtbnMgPSAobGVuZ3RoID0gMTAsIHByZWZpeCA9ICdjb2x1bW4tJywgcHJvcHM/OiBhbnkpID0+XG4gIEFycmF5LmZyb20oeyBsZW5ndGggfSkubWFwKChfLCBjb2x1bW5JbmRleCkgPT4gKHtcbiAgICAuLi5wcm9wcyxcbiAgICBrZXk6IGAke3ByZWZpeH0ke2NvbHVtbkluZGV4fWAsXG4gICAgZGF0YUtleTogYCR7cHJlZml4fSR7Y29sdW1uSW5kZXh9YCxcbiAgICB0aXRsZTogYENvbHVtbiAke2NvbHVtbkluZGV4fWAsXG4gICAgd2lkdGg6IDE1MCxcbiAgfSkpXG5cbmNvbnN0IGdlbmVyYXRlRGF0YSA9IChcbiAgY29sdW1uczogUmV0dXJuVHlwZTx0eXBlb2YgZ2VuZXJhdGVDb2x1bW5zPixcbiAgbGVuZ3RoID0gMjAwLFxuICBwcmVmaXggPSAncm93LSdcbikgPT5cbiAgQXJyYXkuZnJvbSh7IGxlbmd0aCB9KS5tYXAoKF8sIHJvd0luZGV4KSA9PiB7XG4gICAgcmV0dXJuIGNvbHVtbnMucmVkdWNlKFxuICAgICAgKHJvd0RhdGEsIGNvbHVtbiwgY29sdW1uSW5kZXgpID0+IHtcbiAgICAgICAgcm93RGF0YVtjb2x1bW4uZGF0YUtleV0gPSBgUm93ICR7cm93SW5kZXh9IC0gQ29sICR7Y29sdW1uSW5kZXh9YFxuICAgICAgICByZXR1cm4gcm93RGF0YVxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgaWQ6IGAke3ByZWZpeH0ke3Jvd0luZGV4fWAsXG4gICAgICAgIHBhcmVudElkOiBudWxsLFxuICAgICAgfVxuICAgIClcbiAgfSlcblxuY29uc3QgY29sdW1ucyA9IGdlbmVyYXRlQ29sdW1ucygxMClcbmxldCBkYXRhID0gZ2VuZXJhdGVEYXRhKGNvbHVtbnMsIDIwMClcblxuY29sdW1uc1swXS5maXhlZCA9IHRydWVcbmNvbHVtbnNbMV0uZml4ZWQgPSBUYWJsZVYyRml4ZWREaXIuTEVGVFxuY29sdW1uc1s5XS5maXhlZCA9IFRhYmxlVjJGaXhlZERpci5SSUdIVFxuXG5mb3IgKGxldCBpID0gMDsgaSA8IDM7IGkrKykgY29sdW1uc1tpXS5zb3J0YWJsZSA9IHRydWVcblxuY29uc3Qgc29ydEJ5ID0gcmVmPFNvcnRCeT4oe1xuICBrZXk6ICdjb2x1bW4tMCcsXG4gIG9yZGVyOiBUYWJsZVYyU29ydE9yZGVyLkFTQyxcbn0pXG5cbmNvbnN0IG9uU29ydCA9IChfc29ydEJ5OiBTb3J0QnkpID0+IHtcbiAgZGF0YSA9IGRhdGEucmV2ZXJzZSgpXG4gIHNvcnRCeS52YWx1ZSA9IF9zb3J0Qnlcbn1cbjwvc2NyaXB0PlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWUsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ==)

image

[after](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXRhYmxlLXYyXG4gICAgOmNvbHVtbnM9XCJjb2x1bW5zXCJcbiAgICA6ZGF0YT1cImRhdGFcIlxuICAgIDpzb3J0LWJ5PVwic29ydEJ5XCJcbiAgICA6d2lkdGg9XCI3MDBcIlxuICAgIDpoZWlnaHQ9XCI0MDBcIlxuICAgIGZpeGVkXG4gICAgQGNvbHVtbi1zb3J0PVwib25Tb3J0XCJcbiAgLz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgeyBUYWJsZVYyRml4ZWREaXIsIFRhYmxlVjJTb3J0T3JkZXIgfSBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmltcG9ydCB0eXBlIHsgU29ydEJ5IH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5jb25zdCBnZW5lcmF0ZUNvbHVtbnMgPSAobGVuZ3RoID0gMTAsIHByZWZpeCA9ICdjb2x1bW4tJywgcHJvcHM/OiBhbnkpID0+XG4gIEFycmF5LmZyb20oeyBsZW5ndGggfSkubWFwKChfLCBjb2x1bW5JbmRleCkgPT4gKHtcbiAgICAuLi5wcm9wcyxcbiAgICBrZXk6IGAke3ByZWZpeH0ke2NvbHVtbkluZGV4fWAsXG4gICAgZGF0YUtleTogYCR7cHJlZml4fSR7Y29sdW1uSW5kZXh9YCxcbiAgICB0aXRsZTogYENvbHVtbiAke2NvbHVtbkluZGV4fWAsXG4gICAgd2lkdGg6IDE1MCxcbiAgfSkpXG5cbmNvbnN0IGdlbmVyYXRlRGF0YSA9IChcbiAgY29sdW1uczogUmV0dXJuVHlwZTx0eXBlb2YgZ2VuZXJhdGVDb2x1bW5zPixcbiAgbGVuZ3RoID0gMjAwLFxuICBwcmVmaXggPSAncm93LSdcbikgPT5cbiAgQXJyYXkuZnJvbSh7IGxlbmd0aCB9KS5tYXAoKF8sIHJvd0luZGV4KSA9PiB7XG4gICAgcmV0dXJuIGNvbHVtbnMucmVkdWNlKFxuICAgICAgKHJvd0RhdGEsIGNvbHVtbiwgY29sdW1uSW5kZXgpID0+IHtcbiAgICAgICAgcm93RGF0YVtjb2x1bW4uZGF0YUtleV0gPSBgUm93ICR7cm93SW5kZXh9IC0gQ29sICR7Y29sdW1uSW5kZXh9YFxuICAgICAgICByZXR1cm4gcm93RGF0YVxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgaWQ6IGAke3ByZWZpeH0ke3Jvd0luZGV4fWAsXG4gICAgICAgIHBhcmVudElkOiBudWxsLFxuICAgICAgfVxuICAgIClcbiAgfSlcblxuY29uc3QgY29sdW1ucyA9IGdlbmVyYXRlQ29sdW1ucygxMClcbmxldCBkYXRhID0gZ2VuZXJhdGVEYXRhKGNvbHVtbnMsIDIwMClcblxuY29sdW1uc1swXS5maXhlZCA9IHRydWVcbmNvbHVtbnNbMV0uZml4ZWQgPSBUYWJsZVYyRml4ZWREaXIuTEVGVFxuY29sdW1uc1s5XS5maXhlZCA9IFRhYmxlVjJGaXhlZERpci5SSUdIVFxuXG5mb3IgKGxldCBpID0gMDsgaSA8IDM7IGkrKykgY29sdW1uc1tpXS5zb3J0YWJsZSA9IHRydWVcblxuY29uc3Qgc29ydEJ5ID0gcmVmPFNvcnRCeT4oe1xuICBrZXk6ICdjb2x1bW4tMCcsXG4gIG9yZGVyOiBUYWJsZVYyU29ydE9yZGVyLkFTQyxcbn0pXG5cbmNvbnN0IG9uU29ydCA9IChfc29ydEJ5OiBTb3J0QnkpID0+IHtcbiAgZGF0YSA9IGRhdGEucmV2ZXJzZSgpXG4gIHNvcnRCeS52YWx1ZSA9IF9zb3J0Qnlcbn1cbjwvc2NyaXB0PlxuIiwic3JjL1BsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0xMzU4NC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCJcbiAgfVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZSxcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl1cbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJzcmMvZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgbGluay5ocmVmID0gJ2h0dHBzOi8vcHJldmlldy0xMzU4NC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcydcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgfSlcbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy0xMzU4NC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyJ9fQ==)

image

**Review by ryuhangyeong — APPROVED** [review]

**Review by tolking — APPROVED** [review]

**Pull request merged**